### DISCUSSION TITLE
SysVar Polling Intervall konfigurierbar machen / Option zum Abschalten des Pollings integrieren

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#785] [Ideas]
Das SysVar Polling war ja schon häufig Thema...
- vielen Umsteigern / HA-Neulingen sind die 30s zu lang weil sie noch nicht auf ihre SysVars verzichten können / wollen
- die HA Experten brauchen / nutzen keine SysVars und könnten auf das Polling verzichten

ioBroker (hm-rega) und NodeRed (node-red-contrib-ccu) bieten beide die Möglichkeit den Polling-Intervall selbst zu definieren sowie die Möglichkeit das Polling gänzlich abzuschalten.
![ioBroker_HM_Rega_Polling](https://user-images.githubusercontent.com/54898249/209729451-9347c688-deca-479c-96f9-b050c8e91eb8.JPG)
![NodeRed_CCU_Connection_Polling](https://user-images.githubusercontent.com/54898249/209729484-c8763755-04b2-4736-b4b8-1991efb4e104.JPG)

**Sowas würde mir für die Integration auch gut gefallen!**

Zum Einen würde ich selbst das SysVar-Polling wohl auch komplett ausschalten weil ich die SysVars nicht brauche.
Zum Anderen habe ich wohl ein "Datenmengen-Problem", dazu mal ein kleiner Exkurs:

Ich wollte die Zentrale unseres Ferienhäuschens (LTE-Anbindung 7GB Daten/Monat) angebunden per VPN in den lokalen Homeassistant einbinden. Klappte auch wunderbar, nur kam nach knapp 2 Wochen schon die Warnung das das Datenvolumen zu 85% aufgebraucht ist. Da ich nicht weiß ob der hohe Datenverbrauch von den Events (XMLRPC Push) oder dem Polling kommt würde ich letzteres gerne mal testweise abschalten.

**Comment by SukramJ** [comment]
Bin ich zwar kein Fan von, aber kann man machen.

**Reply by SukramJ** [reply]
Mit version 1.26.0b0 kann man jetzt das scan_interval ändern und auch abschalten.
Das minimale Abstand beträgt jetzt 5s. 
Eine entsprechende Erklärung, das sehr kurze Intervalle eigentlich nicht so toll sind, hab ich in die Doku aufgenommen. Ich hab hier bewusst das Wort _Blödsinn_ vermieden.
Die Empfehlung für Datenaktualität bei Sysvar ist natürlich der on-demand Abruf mit `homematicip_local.fetch_system_variables`.

**Reply by Baxxy13** [reply]
Sehr gut, Danke!
Einen Blick konnte ich schon riskieren, testen aber noch nicht...
`This release of Homematic(IP) Local requires HA version 2023.01.0.dev0 and above`
(will jetzt nicht extra die HA Dev Version installieren)

**Reply by SukramJ** [reply]
Wie ich sehe liest du die Release Notes (nicht).
Die heute veröffentlichte Beta geht natürlich auch.

**Reply by Baxxy13** [reply]
Das kommt davon wenn man x Sachen gleichzeitig macht.
Ich gelobe Besserung. :wink:

> Die heute veröffentlichte Beta geht natürlich auch.

Das verstehe ich jetzt nicht so richtig. 
Die 1.26.0b0 ist doch die "heute veröffentlichte Beta", oder? 
Und die braucht HA 2023.01, was ich aktuell nicht habe.

**Reply by SukramJ** [reply]
1.26.0b0 funktioniert mit HA 2023.1.0b0 (auch heute erschienen).

**Reply by SukramJ** [reply]
Wenn du HAOS verwendendes kannst du auf den Beta Kanal wechseln:
Einstellungen -> System -> Updates -> „drei Punkte oben rechts“

**Comment by Baxxy13** [comment]
Feedback:
Nachdem nun HA 2023.1.0 offiziell ist habe ich geupdated und dazu dann gleich die Integration auf 1.26.0 aktualisiert.

Der einstellbare Poll-Intervall funktioniert wunderbar.
Etwas "verwirrt" war ich beim deaktivieren des SysVar Scans.

Den Haken rausgenommen und das Setup beendet. Anschließend nochmal "kontrolliert", da ist der Haken wieder drin.
Kontrolle auf der Zentrale ergab aber das nicht mehr gepollt wird. Funktioniert also auch.

Da das Datenvolumen für das Ferienhaus wieder aufgeladen ist teste ich das jetzt mal ohne SysVar-Poll.

Da ich des Python-Codes nicht mächtig bin...
Könntest du vielleicht kurz zusammenfassen ob und wie oft Daten von der Zentrale aktiv "abgefragt" werden.

Danke nochmal für die Implementierung! :thumbsup:

**Reply by SukramJ** [reply]
Danke für die Rückmeldung.

> Anschließend nochmal "kontrolliert", 

Hier wird der Default und nicht der alte Wert verwendet.
Zur Kontrlle am Besten die Diagnoseinformationen herunterladen.

Die Daten werden immer einmal beim Start abgefragt, aber nicht an HA weitergereicht.
Danach werden die Daten normalerweise alle 30s abgerufen.

**Reply by Baxxy13** [reply]
Alles klar, Danke.
Ich werde dann mal den Datenverbrauch genauer beobachten.
